ls1intum/Artemis
# `Iris`: Fix cooldown message when chat limit is reached

### Checklist
#### General


#### Client

### Motivation and Context
After reaching the rate limit on IRIS, the cooldown message should display a timer after string interpolation, but it does not. The string interpolation does not function correctly and the variables are still displayed in curly braces.

This PR resolves #7729

### Steps for Testing
Prerequisites:
- 1 Instructor
- 1 Programming Exercise with IRIS enabled

Go to any exercise with IRIS enabled
Send messages and reach the rate limit (e.g. 20 messages)
Verify that warning message includes the remaining amount of time for cooldown and the string is interpolated correctly.

### Testserver States
> [!NOTE]
> These badges show the state of the test servers.
> Green = Currently available, Red = Currently locked

![](https://byob.yarr.is/ls1intum/Artemis/artemis-test1)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test2)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test3)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test4)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test5)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test6)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test9)

### Review Progress






#### Performance Review
#### Code Review
#### Manual Tests
#### Exam Mode Test

### Screenshots
Before:
<img width="324" alt="288065078-1a90c09c-c931-4ddf-9358-d570fe129507" src="https://github.com/ls1intum/Artemis/assets/38523756/3980ebba-ef6c-41c6-a062-5455fccae02b">
After:
<img width="329" alt="Screenshot 2023-12-11 at 14 25 30" src="https://github.com/ls1intum/Artemis/assets/38523756/c3dd25a5-c1c1-4ca8-b0a8-0c821a0d176c">
